WordPress/gutenberg
# [Mobile] - Block selection - Expand tapping on nested blocks directly

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5781
- https://github.com/wordpress-mobile/WordPress-iOS/pull/20683
- https://github.com/wordpress-mobile/WordPress-Android/pull/18448

## What?
This PR enables tapping on nested blocks directly as a follow-up of previous code changes where it was enabled for text-based blocks.

## Why?
To keep consistency with the web editor so both have the same behavior when tapping on blocks.

## How?
By updating the `onFocus` code to use the `clientId` instead of looking for the first block to select depending on the nested level.

## Testing Instructions

- Open the editor (There are builds linked above)
- Nest some blocks (try using blocks that are not rich text fields, e.g Galleries, Columns with Images, Spacer, etc.
- Tap on nested blocks
- **Expect** to see the block gain focus directly instead of having to tap several times depending on how many nested levels there are.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 

Before | After
-|-
<video src="https://github.com/WordPress/gutenberg/assets/4885740/5f1da722-2440-4098-a009-aafb07a12884" width=200 /> | <video src="https://github.com/WordPress/gutenberg/assets/4885740/2622e881-989d-4759-813c-96c135f2569c" width=200 />
